src/background/executor.ts

```diff
@@ -33,9 +33,13 @@ import { SessionMap } from "@/mv3/SessionStorage";
 import { groupPromisesByStatus } from "@/utils/promiseUtils";
 import { TOP_LEVEL_FRAME_ID } from "@/domConstants";
 import { forEachTab } from "@/utils/extensionUtils";
+import reportEvent from "@/telemetry/reportEvent";
+import { Events } from "@/telemetry/events";
 
 type TabId = number;
 
+const LARGE_AMOUNT_OF_TABS = 20;
```

Comment:
Can we provide a comment with context on this number? Is this the number after which we think the number of tabs might impact user-visible mod performance? How did we arrive at 20?

Reply:
It's an arbitrary number. I don’t think anything lower than this has any visible effect on performance, but anything above it might be worth tracking. The event itself will include the tab amount.

Would this comment be useful?

```suggestion
// Arbitrary number of tabs above which performance *might* be degraded
const LARGE_AMOUNT_OF_TABS = 20;
```

Reply:
> Would this comment be useful?

IMO yes